File: server/models/reservation.go
package models

import "time"

type Reservation struct {
	ID         uint       `gorm:"primaryKey" json:"id"`
	UserID     uint       `json:"userId"`
	User       User       `gorm:"foreignKey:UserID"`
	CourtID    uint       `json:"courtId"`
	Court      Court      `gorm:"foreignKey:CourtID"`
	Date       time.Time  `json:"date"`
	TimeSlotID uint       `json:"timeSlotId"`
	Timeslot   Timeslot   `gorm:"foreignKey:TimeSlotID"`
	Status     string     `json:"status"` // pending, confirmed, canceled, completed
	Payment    *Payment   `gorm:"foreignKey:ReservationID"`
	CreatedAt  time.Time  `json:"createdAt"`
	UpdatedAt  time.Time  `json:"updatedAt"`
}

File: client/src/types/reservation.ts
import { Court } from "./court";
import { Timeslot } from "./timeslot";
import { Payment } from "./payment";
import { User } from "./user";

export interface Reservation {
  id: number,
  user_id: number,
  court_id: number,
  date: string,
  time_slot_id: number,
  status: string,

  user: User,
  court: Court,
  timeslot: Timeslot,
  payment: Payment
}